### ISSUE TITLE
[FR] Show parameters / tags in metric chart legends

### ISSUE THREAD

**Issue by shaperilio** [mlflow/mlflow#17414]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

Chart legends seem to show either the run name or the run name with metric name in parentheses, e.g.:

<img width="547" height="484" alt="Image" src="https://github.com/user-attachments/assets/8fe4112c-a789-4b4e-8466-8255e33e7190" />

I propose to add an item to the "sprocket" icon where the X-axis can be chosen to be any metric, and provide an option there to choose a parameter or tag to set or add to the legend.

### Motivation

> #### What is the use case for this feature?
It is useful to know if a particular parameter is varying when looking at results in a chart.

> #### Why is this use case valuable to support for MLflow users in general?
I would guess I'm not the only one who does hyperparameter tuning this way :)

> #### Why is this use case valuable to support for your project(s) or organization?
I find myself naming runs according to some parameter value, so when I look at charts, I can be reminded of what I was varying in this particular set of runs. I would rather not do that, and it's often not feasible when several parameters are being varied.

> #### Why is it currently difficult to achieve this use case?
It is not possible to change legend labels.

### What machine learning domain(s) is this feature request about?

- [ ] `domain/genai`: LLMs, Agents, and other GenAI-related use cases
- [ ] `domain/classical-ml`: Traditional machine learning, such as linear regression.
- [ ] `domain/deep-learning`: Deep learning and neural networks.
- [ ] `domain/platform`: MLflow platform foundation, not specific to a particular machine learning domain.

### What area(s) of MLflow is this feature request about?

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

**Comment by daniellok-db** [comment]
i think you can already do this ("Configure" in the three-dot menu), unless i'm misunderstanding the feature request:

<img width="402" height="387" alt="Image" src="https://github.com/user-attachments/assets/2d4cab35-95df-47d2-84fa-d92dd982bc5c" />

<img width="918" height="891" alt="Image" src="https://github.com/user-attachments/assets/f5ba688f-d61e-49bf-9d4c-79555e584f3a" />

**Comment by daniellok-db** [comment]
ah is the request specifically about configuring the legend? could you clarify what your use-case is, and what you'd expect the end result to look like?

**Comment by shaperilio** [comment]
Correct, the request is to modify the legend label text, and it could be set in a similar way that the X-axis metric is set (as shown in your screenshot).

Here's a sketch of what I mean:

<img width="1084" height="1288" alt="Image" src="https://github.com/user-attachments/assets/d3edc724-c074-4fed-93d4-77482b157607" />

There would be an option for "Legend" which could be a tag or parameter value or the run name.

If a parameter or tag is chosen, the name of the parameter or tag becomes the legend title and the values become the label for each trace.

**Comment by shaperilio** [comment]
Now that I think of it, the legend labels could also be a metric.
